### PR TITLE
add in seller name

### LINKED ISSUE #7: Expose seller names
We need seller names for emailing buyer receipts (https://github.com/mozilla/payments-service/issues/58)

### LINKED ISSUE #8: Get seller for product more easily
It would be nice if we had a method to look up the seller for a product. When doing things like emailing receipts (https://github.com/mozilla/payments-service/issues/58) we will only have the product ID (`public_id`) with which to get the product configuration. We could make a map of all sellers keyed by product IDs but that's not so nice.

### PR DESCRIPTION
**Pull request by andymckay**
- helps with mozilla/payments-service#58
- fixes #7 
- fixes #8

### CODE REVIEW

**Comment by andymckay** [comment]
fixes #7

**Comment by andymckay** [comment]
This also fixes #8

Its there in the JSON, but also:

```
>>> from payments_config import products
>>> products['mozilla-concrete-brick']  
<payments_config.Product object at 0x10a51ba10>
>>> products['mozilla-concrete-brick'].seller
<payments_config.Seller object at 0x10a51b950>
```

File: payments_config/__init__.py [hunk 1]
```diff
@@ -69,7 +80,14 @@ def format_prices(self, locales):
         return prices
 
     def to_dump(self):
-        return self.__dict__
+        """
+        When dumping the product, dump the seller as well, but not the
+        products (removed in seller.to_dump), otherwise you'll be in a
+        recursive mess.
```
Comment (kumar303) [inline payments_config/__init__.py:86]: this seems more like a comment than a docstring because it's not explaining the interface of `to_dump()`.

**Comment by kumar303** [comment]
duplicating seller in the JSON is going to start to bloat the bundle in JavaScript land (where space maters). However, I think we need to [rethink the JS lib altogether](https://github.com/mozilla/payments-config/issues/4) so we can sort it out later.

**Comment by kumar303** [comment]
r+wc

**Pull request merged**